### PR TITLE
Create `FUUID` macro annotation to auto-generate `Id` types 

### PR DESCRIPTION
**Pull request by alejandrohdezma**
# What has been done in this PR?

- Create a `@FUUID` annotation that can be used on any kind of object to automatically create an inner `Id` tagged `FUUID` type with convenient methods for its creation. It also provides implicit instances for cats' `Show`, `Hash` and `Order`.

> **Example:**
>
>```scala
>//This...
>@FUUID object Institution
>
>//...is transformed into this...
>object Institution {
>
>	trait IdTag
>      
>	type Id = shapeless.tag.@@[FUUID, IdTag]
>    
>	object Id {
>    	
>		//Creates a new `Id` from a `FUUID`
>		def apply(fuuid: FUUID): Institution.Id =
>			shapeless.tag[IdTag][FUUID](fuuid)
>    
>		//Creates a new `Id` from an `UUID` literal. This method
>		//uses a macro to compile check the literal value
>		def apply(s: String): Institution.Id =
>			macro FUUIDMacros.fuuidLiteral
>    
>		//Creates a random `Id` wrapped in an `F`
>		def random[F[_]: Sync]: F[Institution.Id] =
>			FUUID.randomFUUID[F].map(apply)
>	
>		object Unsafe {
>       
>			//Creates an unwrapped random `Id`
>			def random: Institution.Id =
>				random.unsafeRunSync())
>
> 		}
>        
>	}
>
>	implicit val IdHashOrderShowInstances: Hash[Institution.Id] with Order[Institution.Id] with Show[Institution.Id] = ???
>
>}
>```

# IntelliJ IDEA users

If you are trying to review this PR using IntelliJ IDEA IDE you will probable be thinking that the `FUUID` annotation is not working, since all references to `Id` appear red:

Oh no!

**No problem at all!** This is caused by a [known issue](https://youtrack.jetbrains.com/issue/SCL-8834) in IntelliJ IDEA Scala plugin. But it's easily solved with the help of a small plugin, and, guess what? This PR comes with its own support plugin. Just:

- [Download the plugin](https://transfer.sh/12UE1g/fuuid-idea.jar)
- Inside `IntelliJ Preferences` -> `Plugins` -> `...` -> `Install plugin from disk  ...` and select the downloaded `jar`

And all those nasty red lines will disappear:

Oh yeah!

# References

- [Macro Annotations](https://docs.scala-lang.org/overviews/macros/annotations.html)
- [FUUID](https://github.com/ChristopherDavenport/fuuid)

# Future work

- Contribute the IntelliJ IDEA plugin as another module

### CODE REVIEW

File: core/src/main/scala/io/chrisdavenport/fuuid/annotation/FUUID.scala [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+ *
+ *    }
+ *
+ *    implicit val IdHashOrderShowInstances: Hash[$name.Id] with Order[$name.Id] with Show[$name.Id] = ???
+ *
+ * }
+ * }}}
+ */
+@compileTimeOnly("enable macro paradise to expand macro annotations")
+class FUUID extends StaticAnnotation {
```
Comment (ChristopherDavenport) [inline core/src/main/scala/io/chrisdavenport/fuuid/annotation/FUUID.scala]: Will this conflict on import?

Comment (alejandrohdezma) [reply core/src/main/scala/io/chrisdavenport/fuuid/annotation/FUUID.scala]: Mmmm, with [`FUUID`](https://github.com/ChristopherDavenport/fuuid/blob/880ccfffe028c6038293da2b67a06f0153a23041/modules/core/src/main/scala/io/chrisdavenport/fuuid/FUUID.scala#L11), yep.

Maybe we can think of another name for the annotation... `DeriveId`?

Comment (ChristopherDavenport) [reply core/src/main/scala/io/chrisdavenport/fuuid/annotation/FUUID.scala]: Yeah, that seems reasonable.

Comment (alejandrohdezma) [reply core/src/main/scala/io/chrisdavenport/fuuid/annotation/FUUID.scala]: Updated to `@DeriveId` https://github.com/ChristopherDavenport/fuuid-annotation/pull/1/commits/ef5aabba83a7541273235515c9ee3dfeeb9f3d74

**Comment by ChristopherDavenport** [comment]
Any objections to getting a base out?

**Comment by alejandrohdezma** [comment]
Nope :)

**Pull request merged**